### PR TITLE
Feature/cldn 1968 (Display JSON data inline)

### PR DESCRIPTION
**Pull request by cccs-Dustin**
This PR incorporates the changes requested in the CLDN-2000 ticket ([link](https://cccs.atlassian.net/browse/CLDN-2000))

**Major Changes Include:**
- Updated version of the `CCCS-Grid` which now displays JSON data inline
- Modified multiple files when running the `pre-commit` hook

### CODE REVIEW

**Review by cccs-RyanK — APPROVED** [review]

**Review by cccs-rc — CHANGES_REQUESTED** [review]
Requesting changes just for the TODOs. I'm not too concerned about the other nitpicks.

File: superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx [hunk 1]
```diff
@@ -454,7 +452,7 @@ const config: ControlPanelConfig = {
                     return newState;
                   },
                   visibility: ({ controls }) =>
-                    // TODO properly emsure is Bool
+                    // TODO properly ensure is Bool
```
[the remark this answers is not in the document]
Comment (cccs-Dustin) [reply superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx]: Removed the 'TODO's in the latest commit

File: superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx [hunk 2]
```diff
@@ -590,8 +588,11 @@ config.controlPanelSections.push({
           renderTrigger: true,
           default: false,
           description: t(
-            'Whether to enable row grouping (this will only take affect after a save)',
+            'Whether to enable row grouping (this will only take affect after a save). NOTE: "JSON Row Expand" and "Row Grouping" are mutually exclusive. If "Row Grouping" is selected, "JSON Row Expand" will not be visible.',
           ),
+          visibility: ({ controls }) =>
+            // TODO properly ensure is Bool
```
[the remark this answers is not in the document]
Comment (cccs-Dustin) [reply superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx]: Removed the 'TODO's in the latest commit

File: superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx [hunk 3]
```diff
@@ -607,6 +608,23 @@ config.controlPanelSections.push({
...
+        config: {
+          type: 'CheckboxControl',
+          label: t('JSON Row Expand'),
+          renderTrigger: true,
+          default: false,
+          description: t(
+            'Whether to enable row level JSON expand buttons. NOTE: "JSON Row Expand" and "Row Grouping" are mutually exclusive. If "JSON Row Expand" is selected, "Row Grouping" will not be visible.',
+          ),
+          visibility: ({ controls }) =>
+            // TODO properly ensure is Bool
```
[the remark this answers is not in the document]
Comment (cccs-Dustin) [reply superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/plugin/controlPanel.tsx]: Removed the 'TODO's in the latest commit

File: superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/ExpandAllValueRenderer.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  };
+
+  // Get all of the cells in the AG Grid and only keep the ones
+  // that are in the same row as the current expand all button and
+  // make sure that they have a JSON blob (and see whether they are
+  // expanded or not)
+  checkState = () => {
+    const newInstances = this.getJSONCells();
+
+    const jsonCells = newInstances.map((instance: any) =>
```
[the remark this answers is not in the document]
Comment (cccs-Dustin) [reply superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/ExpandAllValueRenderer.tsx:101]: Ranamed the variable to be equal to `jsonCellExpandedValues` in the most recent commit.

File: superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/ExpandAllValueRenderer.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      this.setState(prevState => ({
+        ...prevState,
+        expanded: false,
+      }));
+    } else {
+      this.setState(prevState => ({
+        ...prevState,
+        expanded: true,
+      }));
+    }
```
[the remark this answers is not in the document]
Comment (cccs-Dustin) [reply superset-frontend/src/cccs-viz/plugins/plugin-chart-cccs-grid/src/ExpandAllValueRenderer.tsx]: Not too sure why the pre-commit hook didn't pick this up. I added the change (to multiple locations) in one of the most recent commits.

**Review by cccs-rc — APPROVED** [review]

**Review by cccs-RyanS — APPROVED** [review]

**Pull request merged**